frontend/src/components/Tag.tsx

```diff
@@ -0,0 +1,73 @@
+import styled from "@emotion/styled";
+
+const getColor = (id: string | number) => {
+  let sum = 0;
+
+  if (typeof id === "number") {
+    sum = id;
+  } else {
+    // compute some deterministic number of the id
+
+    //iterate over the string and get the sum of the char codes
+
+    for (let i = 0; i < id.length; i++) {
+      sum += id.charCodeAt(i);
+    }
+  }
+
+  // now mod it by the number of colors
+  return sum % 8;
+};
```

Comment:
this simple function isn't immediately obvious, and it's simple enough that it should be

```suggestion
// deterministically get a color for any id
const getColor = (id: string | number) => {
  const colorCount = 8;

  if (typeof id === "number") {
    return id % colorCount;
  }
  
  // compute some deterministic number from the id
  let sum = 0;
  for (let i = 0; i < id.length; i++) {
    sum += id.charCodeAt(i);
  }

  return sum % colorCount;
};
```

by ensuring `sum` is only declared where we need it, that comments don't repeat the code, etc. you get something much cleaner.
Arguably `colorCount` should then be moved outside the function.